components/service/glean/src/main/java/mozilla/components/service/glean/Dispatchers.kt

```diff
@@ -64,6 +86,73 @@ internal object Dispatchers {
         fun setTestingMode(enabled: Boolean) {
             testingMode = enabled
         }
+
+        /**
+         * Enable queueing mode, which causes tasks to be queued until launched by calling
+         * [flushQueuedInitialTasks].
+         *
+         * @param enabled whether or not to enable the testing mode
+         */
+        @VisibleForTesting(otherwise = VisibleForTesting.NONE)
+        fun setTaskQueueing(enabled: Boolean) {
+            queueInitialTasks = enabled
+        }
+
+        /**
+         * Stops queueing tasks and processes any tasks in the queue. Since [queueInitialTasks] is
+         * set to false prior to processing the queue, newly launched tasks should be executed
+         * on the couroutine scope rather than added to the queue.
+         */
+        internal fun flushQueuedInitialTasks() {
+            // Setting this to false first should cause any new tasks to just be executed (see
+            // launch() above) making it safer to process the queue.
```

Comment:
nit: please extend this comment to mention that flipping this could cause some tasks to be executed out of order (explain the reason) and also mention the bug that its looking to fix this.